src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs

```diff
@@ -1969,7 +1969,7 @@ protected bool TryGetInterproceduralAnalysisResult(IOperation operation, out TAn
             }
 
             // Check if we are already at the maximum allowed interprocedural call chain length.
-            int currentMethodCallCount = currentMethodsBeingAnalyzed.Where(m => !((IMethodSymbol)m.OwningSymbol).IsLambdaOrLocalFunctionOrDelegate()).Count();
+            int currentMethodCallCount = currentMethodsBeingAnalyzed.Where(m => !(m.OwningSymbol is IMethodSymbol ms && ms.IsLambdaOrLocalFunctionOrDelegate())).Count();
```

Comment:
OwningSymbol is no longer just an IMethodSymbol, since DFA is also looking at field / property initializers.  Since we're just excluding lambdas / locals / delegates from the count, I'm hoping this is okay.